### PR TITLE
feat: implemented source_simple_resonator

### PR DESCRIPTION
**Pull request by Anshgrover23**
closes tscircuit/core#426
/claim #426

### CODE REVIEW

File: src/source/source_simple_resonator.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { source_component_base } from "src/source/base/source_component_base"
+import { z } from "zod"
+
+export const source_simple_resonator = source_component_base.extend({
+  ftype: z.literal("resonator"),
+  frequency: z.union([z.string(), z.number()]),
+  load_capacitance: z.union([z.string(), z.number()]),
+  pin_count: z.number().default(3), // Set the default pin count to 3 for resonators
+  gender: z.enum(["male", "female"]).optional().default("male"),
```
Comment (imrishabh18) [inline src/source/source_simple_resonator.ts:9]: I don't think we will need gender for resonator.

cc: @seveibar

**Comment by Anshgrover23** [comment]
@imrishabh18 so i  will remove that one.

File: src/source/source_simple_resonator.ts (hunk 1 above)
Comment (seveibar) [reply src/source/source_simple_resonator.ts]: nice catch @imrishabh18

File: src/source/source_simple_resonator.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { source_component_base } from "src/source/base/source_component_base"
+import { z } from "zod"
+
+export const source_simple_resonator = source_component_base.extend({
+  ftype: z.literal("simple_resonator"),
+  frequency: z.union([z.string(), z.number()]),
```
Comment (seveibar) [inline src/source/source_simple_resonator.ts:6]: this should be "frequency", if it's not defined it should be a well-defined unit

Comment (seveibar) [reply src/source/source_simple_resonator.ts:6]: CC @imrishabh18 this is a tricky one to catch

Comment (seveibar) [reply src/source/source_simple_resonator.ts:6]: "1Hz" is a valid frequency for example, it's parsed into a standard unit `Hz`

**Review by seveibar — CHANGES_REQUESTED** [review]
see frequency issue

File: src/source/source_simple_resonator.ts (hunk 2 above)
Comment (Anshgrover23) [reply src/source/source_simple_resonator.ts]: done.

**Comment by Anshgrover23** [comment]
@seveibar done.

File: src/source/source_simple_resonator.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+import { frequency } from "src/units"
+
+export const source_simple_resonator = source_component_base.extend({
+  ftype: z.literal("simple_resonator"),
+  frequency,
+  load_capacitance: z.union([z.string(), z.number()]),
+  pin_count: z.number().default(3), // Set the default pin count to 3 for resonators
+})
+
+export type SourceSimpleResonator = z.infer<typeof source_simple_resonator>
```
Comment (seveibar) [inline src/source/source_simple_resonator.ts:12]: this should be defined as an interface and use `expectTypesMatch`

**Review by seveibar — CHANGES_REQUESTED** [review]
didn't catch this the first time, but check out the new pattern for expectTypesMatch, the scripts can't generate documentation if you use the old pattern

File: src/source/source_simple_resonator.ts (hunk 3 above)
Comment (Anshgrover23) [reply src/source/source_simple_resonator.ts]: done.

**Review by seveibar — CHANGES_REQUESTED** [review]
load_capacitance is a unit too

File: src/source/source_simple_resonator.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  source_component_base,
+  type SourceComponentBase,
+} from "src/source/base/source_component_base"
+import { frequency, capacitance } from "src/units"
+import { expectTypesMatch } from "src/utils/expect-types-match"
+
+export const source_simple_resonator = source_component_base.extend({
+  ftype: z.literal("simple_resonator"),
+  frequency,
+  capacitance,
```
Comment (seveibar) [inline src/source/source_simple_resonator.ts:12]:
```suggestion
  load_capacitance,
```

File: src/source/source_simple_resonator.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  pin_count: z.number().default(3), // Set the default pin count to 3 for resonators
```
Comment (seveibar) [inline src/source/source_simple_resonator.ts:13]:
```suggestion
```

File: src/source/source_simple_resonator.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+type InferredSourceSimpleResonator = z.infer<typeof source_simple_resonator>
+
+/**
+ * Defines a simple resonator component
+ */
+export interface SourceSimpleResonator extends SourceComponentBase {
+  ftype: "simple_resonator"
+  frequency: number
+  capacitance: number
+  pin_count: number
```
Comment (seveibar) [inline src/source/source_simple_resonator.ts:26]:
```suggestion
```

File: src/source/source_simple_resonator.ts [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+import {
+  source_component_base,
+  type SourceComponentBase,
+} from "src/source/base/source_component_base"
+import { frequency, capacitance } from "src/units"
+import { expectTypesMatch } from "src/utils/expect-types-match"
+
+export const source_simple_resonator = source_component_base.extend({
+  ftype: z.literal("simple_resonator"),
+  frequency,
```
Comment (seveibar) [inline src/source/source_simple_resonator.ts:11]:
```suggestion
  frequency,
  equivalent_series_resistance: resistance.optional(),
```

File: src/source/source_simple_resonator.ts [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+export type SourceSimpleResonatorInput = z.input<typeof source_simple_resonator>
+type InferredSourceSimpleResonator = z.infer<typeof source_simple_resonator>
+
+/**
+ * Defines a simple resonator component
+ */
+export interface SourceSimpleResonator extends SourceComponentBase {
+  ftype: "simple_resonator"
+  frequency: number
+  capacitance: number
```
[the remark this answers is not in the document]
Comment (Anshgrover23) [reply src/source/source_simple_resonator.ts:25]: done.

File: src/source/source_simple_resonator.ts (hunk 4 above)
Comment (Anshgrover23) [reply src/source/source_simple_resonator.ts:12]: done.

File: src/source/source_simple_resonator.ts (hunk 5 above)
Comment (Anshgrover23) [reply src/source/source_simple_resonator.ts:13]: sorry its pinVariant

File: src/source/source_simple_resonator.ts (hunk 7 above)
Comment (Anshgrover23) [reply src/source/source_simple_resonator.ts:11]: applied in optional

File: src/source/source_simple_capacitor.ts [hunk 9]
```diff
@@ -3,12 +3,15 @@ import {
...
 } from "src/source/base/source_component_base"
-import { capacitance } from "src/units"
+import { capacitance, resistance, frequency } from "src/units"
 import { expectTypesMatch } from "src/utils/expect-types-match"
 
 export const source_simple_capacitor = source_component_base.extend({
   ftype: z.literal("simple_capacitor"),
-  capacitance,
+  load_capacitance: capacitance,
+  pinVariant: z.literal("3pin"),
```
Comment (seveibar) [inline src/source/source_simple_capacitor.ts:12]:
```suggestion
```

File: src/source/source_simple_capacitor.ts [hunk 10]
```diff
@@ -19,7 +22,10 @@ type InferredSourceSimpleCapacitor = z.infer<typeof source_simple_capacitor>
  */
 export interface SourceSimpleCapacitor extends SourceComponentBase {
   ftype: "simple_capacitor"
-  capacitance: number
+  load_capacitance: number
+  pinVariant: "3pin"
```
Comment (seveibar) [inline src/source/source_simple_capacitor.ts:26]:
```suggestion
```

**Review by seveibar — CHANGES_REQUESTED** [review]
don't change capacitor file

**Review by seveibar — APPROVED** [review]

**Pull request merged**